src/main/kotlin/it/pagopa/ecommerce/eventdispatcher/services/RefundService.kt

```diff
@@ -63,4 +82,69 @@ class RefundService(
   fun requestXpayRefund(requestID: String): Mono<XPayRefundResponse200Dto> {
     return paymentGatewayClient.requestXPayRefund(UUID.fromString(requestID))
   }
+
+  fun requestRedirectRefund(
+    transactionId: TransactionId,
+    pspTransactionId: String,
+    paymentTypeCode: String,
+    pspId: String
+  ): Mono<RedirectRefundResponseDto> =
+    getRedirectUri(pspId, paymentTypeCode)
+      .fold(
+        { Mono.error(it) },
+        { uri ->
+          logger.info(
+            "Redirect transaction refund. TransactionId: [{}], pspTransactionId: [{}], payment type code: [{}}",
+            transactionId,
+            pspTransactionId,
+            paymentTypeCode)
+          nodeForwarderRedirectApiClient
+            .proxyRequest(
+              RedirectRefundRequestDto()
+                .action("refund")
+                .idPSPTransaction(pspTransactionId)
+                .idTransaction(transactionId.value()),
+              uri,
+              transactionId.value(),
+              RedirectRefundResponseDto::class.java)
+            .onErrorMap(NodeForwarderClientException::class.java) { exception ->
+              val errorCause = exception.cause
+              val httpErrorCode: Optional<HttpStatus> =
+                Optional.ofNullable(errorCause).map {
+                  if (it is WebClientResponseException) {
+                    it.statusCode
+                  } else {
+                    null
+                  }
+                }
+              httpErrorCode
```

Comment:
What about adding a `log.error`? This would make log consultation easier

Reply:
You're absolutely right! Logs reviewed with https://github.com/pagopa/pagopa-ecommerce-event-dispatcher-service/pull/239/commits/76601d311abf4528c0a338761eb26a532b18061c